### PR TITLE
Feature/cmake

### PR DESCRIPTION
**Pull request by lizziel**
Updates to enable building HEMCO as submodules within GCHPctm and the new GEOS-Chem Classic wrapper. These updates need review and discussion prior to merging.

### CODE REVIEW

**Comment by LiamBindle** [comment]
Also, I don't think the try_compile code is necessary in HEMCO. That code was really just UX code for GEOS-Chem Classic to error during the CMake step if the user's NetCDF installation was broken. I think for HEMCO it's unnecessary complexity.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,231 +1,264 @@
...
-#   SRCS ${srcs}
-#   DEPENDENCIES MAPL_Base
-#   )
-#
-#target_include_directories (${this} PUBLIC ${INC_ESMF} ${INC_NETCDF})
-#
-#target_compile_definitions (${this} PRIVATE ESMF_ DEVEL NO_BPCH UCX
-#   EXTERNAL_GRID EXTERNAL_FORCING GEOS_FP USE_REAL8 USE_TEND NC_DIAG DISCOVER MODEL_GEOS)
-#set_target_properties (${this} PROPERTIES Fortran_MODULE_DIRECTORY ${include_${this}})
+
```
Comment (LiamBindle) [inline CMakeLists.txt]: I don't think the try_compile code is necessary in HEMCO. That code was really just UX code for GEOS-Chem Classic to error during the CMake step if the user's NetCDF installation was broken. I think for HEMCO it's unnecessary complexity.

Comment (LiamBindle) [reply CMakeLists.txt]: This remove the need for CMakeScripts/try_compile.F90

**Comment by lizziel** [comment]
Thanks @LiamBindle . Regarding inclusion of try_compile, I wonder if we would still want it for the case of HEMCO standalone. If the netcdf installation is broken and the try_compile step is removed, does the error ultimately happen at run-time?

**Comment by LiamBindle** [comment]
@lizziel The error would be at compile time. Keeping the try_compile seems reasonable to me. I think the thing we should push is: if the configure step passes and the build step fails then that's a bug in the CMakeLists. The try_compile is useful for throwing an error during the configure step, so I think it's reasonable to keep it.

**Comment by lizziel** [comment]
I made the suggested changes in new commit https://github.com/geoschem/HEMCO/pull/23/commits/a4bfaaba665e782bb6fd2ee767956c0f18c61c86.

**Comment by LiamBindle** [comment]
Looks good to me!

**Pull request merged**